Alejandro Camarillo's PR

tribakzero opened the pull request.

tribakzero committed:
Git Ignore for Visual Studio Code

tribakzero committed:
feat: Adds home and favorites routes and components

tribakzero committed:
feat: Added more components and cleaned the code in general

tribakzero committed:
feat: Linked Home to Giphy API and saved to redux, also modified for new gif API

tribakzero committed:
feat: Add, remove and list favorites, added. Removed API_KEY

alejandro-krieger committed:
feat: Home Search on /search Giphy endpoint

tribakzero committed:
chore: Added DEVTOOLS for Redux

tribakzero committed:
feat: Adds favorite filtering

tribakzero committed:
feat: Refactor fetcher, replaced with sagas

tribakzero committed:
refactor: Moved sagas to own files

alejandro-krieger committed:
Adds empty styled components, new components structure and uses aliases to prevent multiple ../

Commit:
fix: Bug on favorites search
Prevented the search results to be removed from the search if they were clicked

Commit:
.env alias instructions

Commit:
First unit tests

tribakzero committed:
wip: Unit testing

alejandro-krieger committed:
fix: Makes big general changes.
Removes css files.
Removes useless container HTML elements.
Adds Fragments
Styles using Styled Components.
Adds attr to styled component.
Adjusts gif size and format.
Changes sopies.
copies*.

bgilm commented on .env.example:
```diff
@@ -0,0 +1,5 @@
+# Rename this file to .env and edit the REACT_APP_API_KEY value with your own key.
```
👍

tribakzero replied:
🆒

bgilm commented on src/App.js:
```diff
@@ -1,19 +1,23 @@
+      <Provider store={store}>
+        <div className="App">
+          <Header />
+          <Switch>
```
👍

tribakzero replied:
🆒

bgilm commented on src/App.js:
```diff
@@ -1,19 +1,23 @@
+          <Switch>
+            <Route exact path="/" component={Home} />
+            <Route path="/favorites" component={Favorites} />
+          </Switch>
```
It would be nice to handle not found routes and either redirect the user to a known route or to a specific component letting him know what happened

tribakzero replied:
Nice one, I just updated it to include the Redirect, thanks 😃

bgilm commented on src/components/favorites/Favorites.js:
```diff
@@ -0,0 +1,65 @@
+    )}
+
+    return (
+      <React.Fragment>
```
👍

tribakzero replied:
🆒

bgilm commented on src/components/header/Header.js:
```diff
@@ -0,0 +1,14 @@
+const Header = () => {
+  return (
+    <StyledHeader>
+      <NavLink to="/" exact>Home</NavLink>
```
👍

tribakzero replied:
🆒

bgilm commented on src/components/home/Home.js:
```diff
@@ -0,0 +1,66 @@
+    }
+  }
+
+  componentDidMount = () => {
```
👍

tribakzero replied:
🆒

timgivois commented on src/App.js:
```diff
@@ -1,19 +1,23 @@
-        </p>
-      </div>
+      <Provider store={store}>
+        <div className="App">
```
you don't need classNames if you use styled-components

tribakzero replied:
I thought this one was related to how React handles the root, but removed it and still works, cool! Thanks.

timgivois commented on src/components/gif/Gif.js:
```diff
@@ -0,0 +1,41 @@
+    return  `#${(Math.floor(Math.random()*16777215).toString(16))}`;
+  }
+  return (
+    <StyledGif bgColor={bgColor} className="gif" src={props.src} onClick={props.onClick}>
```
why are you using classNames? :'(

tribakzero replied:
Sorry mate hahaha, it's solved now.

timgivois commented on src/components/gif/Gif.style.js:
```diff
@@ -0,0 +1,21 @@
+  display: flex;
+  min-width: 66px;
+  position: relative;
+  .like-status {
```
avoid using classnames, create components instead

tribakzero replied:
Updated as suggested.

timgivois commented on src/components/gifList/GifList.style.js:
```diff
@@ -0,0 +1,11 @@
+  display: flex;
+  flex-wrap: wrap;
+  justify-content: center;
+  .gif {
```
create another component

tribakzero replied:
Hey Tim, thank you for your feedback, I updated as suggested.

alejandro-krieger committed:
feat: Empty placeholders for searchs and favorites

alejandro-krieger committed:
feat: Updates according to reviews.
feat: Adds redirect to Home if Route is missing.
fix: Moves class based styles to styled components.
fix: Removes useless wrappers and changes as needed for Fragments.

timgivois commented on src/components/gif/Gif.js:
```diff
@@ -0,0 +1,41 @@
+  return (
+    <StyledGif bgColor={bgColor} src={props.src} onClick={props.onClick}>
+      <img
+        className="img"
```
another className compa

timgivois commented on src/components/likeStatus/LikeStatus.js:
```diff
@@ -0,0 +1,10 @@
+import { StyledLikeStatus } from './LikeStatus.style'
+
+const LikeStatus = (props) => {
+  return <StyledLikeStatus className="like-status">
```
:'(

timgivois commented on src/components/searchBar/SearchBar.style.js:
```diff
@@ -0,0 +1,13 @@
+import styled from 'styled-components';
+
+export const SearchBar = styled.div`
+  input {
```
I think that you can just use an input without enveloping a div

timgivois commented on src/components/gif/Gif.style.js:
```diff
@@ -0,0 +1,21 @@
+import styled from 'styled-components';
+
+
+export const StyledGif = styled.div.attrs({
```
🆒

timgivois commented on src/components/header/Header.style.js:
```diff
@@ -0,0 +1,19 @@
+  height: 3em;
+  align-items: center;
+  justify-content: center;
+  a {
```
another component please

timgivois commented on src/globalStyles.js:
```diff
@@ -0,0 +1,11 @@
+import { injectGlobal } from 'styled-components'
+
+injectGlobal`
```
🆒

timgivois commented on src/test/Favorites.test.js:
```diff
@@ -0,0 +1,20 @@
+let wrapper;
+let store;
+
+beforeEach(() => {
```
the beforeEach runs inside thhe describe

timgivois commented on src/test/SearchBar.test.js:
```diff
@@ -0,0 +1,20 @@
+import React from 'react';
+import { shallow } from 'enzyme';
+import SearchBar from 'components/searchBar/SearchBar'
```
is this a container or a component?

timgivois commented on src/test/LikeStatus.test.js:
```diff
@@ -0,0 +1,10 @@
+import React from 'react';
+import { shallow } from 'enzyme';
+import LikeStatus from 'components/likeStatus/LikeStatus'
```
is this a container or a component?

timgivois commented:
ttry to snapshot only components not containers

Commit:
fix: Updates some files for unit testing and styling

tribakzero committed:
fix: Unit tests

tribakzero committed:
fix: Unit tests

timgivois commented:
Travis doesn't have the NODE_PATH. But I could run your tests. It's ok 💯

Robfz commented on src/actions/trending.js:
```diff
@@ -0,0 +1,18 @@
+const TRENDING_GIFS_FETCHED = 'TRENDING_GIFS_FETCHED';
+
+const getTrendingGifs = (payload = {}) => ({ type: TRENDING_GIFS_GET, payload });
+const fetchedTrendingGifs = createAction(TRENDING_GIFS_FETCHED);
```
Any reason why you are mixing the creators declaration between `createAction` and manual?

Robfz commented on src/components/favorites/Favorites.js:
```diff
@@ -0,0 +1,79 @@
+    var filtered = this.filteredFavorites();
+    let searchToggle = null;
+
+    if(!this.props.favorite) return null
```
All the logic regarding filtering can be done in the Store.

Robfz commented on src/components/gifList/GifList.js:
```diff
@@ -0,0 +1,58 @@
+            liked={props.gifs[gif].liked}
+            title={props.gifs[gif].title}
+            src={props.gifs[gif].images.fixed_width_small.url}
+            onClick={() => toggleFavorite(gif)}
```
Nice currying.

Robfz commented on src/components/home/Home.js:
```diff
@@ -0,0 +1,82 @@
+  render() {
+    let searchToggle = null;
+
+    if(!this.props.trending) return null
```
This code is basically the same than before, therefore why could be useful to have this logic in the Redux store.
